projects/can_api/utils.py

```diff
@@ -1,14 +1,21 @@
 def get_rx_messages(subs, messages):
     """
-    Gets the received Messages from the list of messages, based on the subs 
+    Gets the received Messages from the list of messages, based on the subs
     dictionary.
     """
     rx_message_names = [d["name"] for d in subs]
     rx_messages = list(filter(lambda m: m.name in rx_message_names, messages))
+    masks = {message["name"]: message.get("mask", 0x7FF) for message in subs}
 
     mobs = {}
 
     for m in subs:
         mobs[m["name"]] = m["mob"]
 
-    return rx_messages, mobs
+    for mask_subscription_name in masks.keys():
```

Comment:
It makes more sense to iterate through `rx_messages` I think. Same outcome, but more idiomatic.

Reply:
Actually, this isn't doing anything. I don't think this will ever fail because `masks` is created from `subs` and so is `rx_message_names`, so they will always have the same items. This might not be necessary (I think the error might be caught elsewhere).

But you should try making a test for it! Create a test that tries to subscribe to a message not in the database and assert that it fails somehow.

Reply:
Yeah, I think it would make sense to do this check just after line 7, and then check if each member of rx_message_names is in rx_messages (bearing in mind you'd need to to pull out the name for each message object).

I did test locally though, without some type of error handling it'll still work, it'll just not say you have an undefined message and generate masks and mobs without using them.